packages/render-fidelity-tools/src/components/renderers/babylon-viewer.ts

```diff
@@ -135,10 +138,12 @@ export class BabylonViewer extends LitElement {
     // in other renderers
     environment.setReflectionTextureMatrix(
         Matrix.RotationY(Tools.ToRadians(90)));
-    const skybox =
-        this[$scene].createDefaultSkybox(this[$scene].environmentTexture!);
-    skybox!.rotate(Axis.Y, Math.PI * 1.5, Space.WORLD);
-    skybox!.infiniteDistance = true;
+    if (renderSkybox) {
+      const skybox =
+          this[$scene].createDefaultSkybox(this[$scene].environmentTexture!);
+      skybox!.rotate(Axis.Y, Math.PI * 1.5, Space.WORLD);
```

Comment:
I know it was like this before, but I'm confused why you have to do `skybox!`. Is it possible `createDefaultSkybox` returns `null`?

Reply:
That's really a good question and i looked into that after you asked. the only way skybox can be null is when this[$scene].environmentTexture is null. the created environment variable cannot be null (if there's an error on creating that there'll be error and it will just stopped at that point.). It's just when assign environment to this[$scene].environmentTexture, environment is cast from BasetTexture to Nullable<BasetTexture>, that's why i need to put! after it. Don't know why they to this cast. Also, not sure if there's better way to handle null exception here. Do you have any idea about that? Thank you!

Reply:
from BaseTexture to Nullable<BaseTexture>

Reply:
from BaseTexture to Nullable(BaseTexture) , not sure why <>  is not shown correctly

Reply:
Okay, I see. Thanks for the explanation. Maybe we can refactor in the future, in general it's probably best to use `!` as sparingly as possible. But I'm okay with this for now.